src/azul/service/elasticsearch_service.py

```diff
@@ -328,27 +329,47 @@ def _generate_paging_dict(self, es_response, pagination):
         # ...else use search_after/search_before pagination
         es_hits = es_response['hits']['hits']
         count = len(es_hits)
+        next_ = {}
+        previous = {}
         if 'search_before' in pagination:
             # hits are reverse sorted
             if count > pagination['size']:
                 # There is an extra hit, indicating a previous page.
                 count = count - 1
                 search_before = es_hits[count - 1]['sort']
+                previous['search_before'], previous['search_before_uid'] = search_before
             else:
                 # No previous page
                 search_before = [None, None]
             search_after = es_hits[0]['sort']
+            next_['search_after'], next_['search_after_uid'] = search_after
+            pagination.pop('search_before', None)
```

Comment:
Why do you have to pop all this stuff at different times from `pagination`? Would it make more sense to build a new dictionary for the `next_` and `previous` URLs? Maybe even making a function that generates the next / previous URLs would separate out some of the logic and make this cleaner.